src/functional/ffront/foast_passes/closure_var_type_deduction.py

```diff
@@ -0,0 +1,59 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2022, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either vervisit_Constantsion 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from dataclasses import dataclass
+from typing import Any
+
+import functional.ffront.field_operator_ast as foast
+from eve import NodeTranslator, traits
+from functional.ffront.symbol_makers import make_symbol_type_from_value
+
+
+@dataclass(frozen=True)
+class ClosureVarTypeDeduction(NodeTranslator, traits.VisitorWithSymbolTableTrait):
+    """
+    Deduce the type of `Symbol` nodes that introduce closure variables.
+
+    The types are deduced by looking at the values inside the list of
+    closure variables. The types are inferred from the Python value.
+
+    For the general type deduction pass, all Symbol nodes must already be typed,
+    this pass must run before.
+    """
+
+    closure_vars: dict[str, Any]
+
+    @classmethod
+    def apply(cls, node: foast.FieldOperator, closure_vars: dict[str, Any]):
```

Comment:
Return type and argument annotation as above.

Reply:
Should be done, I checked all type annotations.